Makefile

```diff
@@ -50,17 +50,18 @@ build-natives:
 	@env GOOS=darwin GOARCH=$$arch CGO_ENABLED=0 \
 		$(GO) build -a $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) -o $(PKGNAME)-darwin ;
 	@echo "building linux";
-	@env GOOS=linux GOARCH=$$arch \
+	@env GOOS=linux GOARCH=$$arch CGO_ENABLED=0 \
 		$(GO) build -a $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) -o $(PKGNAME)-linux ;
 	@echo "building windows";
 	@env GOOS=windows GOARCH=$$arch CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ \
 		$(GO) build $(GO_LDFLAGS_WIN) $(BUILDV) -tags $(TAGS) nolzma -o $(PKGNAME)-windows.exe ;
 
 build-natives-contained:
 	rm -f mender-artifact && \
-	image_id=$$(docker build -f Dockerfile.binaries . | awk '/Successfully built/{print $$NF;}') && \
-	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact* /binary" && \
-	docker image rm $$image_id
+	docker build -f Dockerfile.binaries -t binary-image . | awk '/Successfully built/{print $$NF;}' && \
+	container_id=$$(docker create binary-image) && \
```

Comment:
I think that if we're using a hardcoded image name, we should keep the mender-artifact name somewhere in it, so it can be recognized later. It also reduces the risk of colliding with a different container.